### ISSUE TITLE
NEI.hide removes all items with same ID when in a loop

### ISSUE THREAD

**Issue by Hazelas81** [stanhebben/MineTweaker3#174]
NEI.Hide, when put into a loop and using an array, will hide *all* instances of an item regardless of metadata specified.

For example, I'm trying to hide WIP Items from the Backpacks mod. All backpacks use the ID of 4172, but use different metadata ranges (0-16, 100-116, and 200-216) to differ between tier and color.

Making an array of all mid-tier backpacks (<Backpack:backpack:100 - 116>), using a for loop, and executing NEI.Hide for this array inside the loop hides *all* backpacks, regardless of meta-data.

Hopefully this is thorough as needed.

**Comment by vanceza** [comment]
1) Could you include sample code?
2) Are you on Minecraft 1.6.4? See the note at the bottom of this page: http://minetweaker3.powerofbytes.com/wiki/Mods:NEI_Support

**Comment by vanceza** [comment]
See also #170

**Comment by Hazelas81** [comment]
1) Sure:
```
//Declaring an Array of items
val smallDyedBackpackArray = [<Backpack:backpack:1>, <Backpack:backpack:2>, <Backpack:backpack:3>, <Backpack:backpack:4>, <Backpack:backpack:5>, <Backpack:backpack:6>, <Backpack:backpack:7>, <Backpack:backpack:8>, <Backpack:backpack:9>, <Backpack:backpack:10>, <Backpack:backpack:11>, <Backpack:backpack:12>, <Backpack:backpack:13>, <Backpack:backpack:14>, <Backpack:backpack:15>, <Backpack:backpack:16>] as IItemStack[];
//Attempting to hide them with a loop
for i, smallDyedBackpack in smallDyedBackpackArray
{
   NEI.Hide(smallDyedBackpack);
}
```

This section of code will hide *all* backpacks (including large backpacks and ender backpacks) as they all share the same ID, but use different meta-data to differentiate between each tier and dye.

2) I am using version 1.7.10.

**Comment by jaredlll08** [comment]
NEI support is broken and cannot be fixed, this issue is resolved in 1.8.x using JEI
